Adding additional config/utility.

Ragundah opened the pull request:
Modified header to be cleaner.
Modified config layout.
Added following config options:
- Config_mailStationary
- Config_maxGMRank
- Config_preventReturn
- Config_charactersDbName

Renamed senderGUID to Config_senderGUID
Added function PreventReturn(playerGUID)
Tested and confirmed viability and functionality of additions and modifications.

Ragundah committed:
Adding additional config/utility.

Modified header to be cleaner.
Modified config layout.
Added following config options:
- Config_mailStationary
- Config_maxGMRank
- Config_preventReturn
- Config_charactersDbName
Renamed senderGUID to Config_senderGUID
Added function PreventReturn(playerGUID)
Tested and confirmed viability and functionality of additions and modifications.

Ragundah committed:
Corrected default Config_senderGUID

Ragundah committed:
Correcting spacing

before playerName = nil lines 172 and 163

Ragundah committed:
Second spacing pass

Fixed spacing on lines: 172, 163, 104, 113

55Honey reviewed (changes requested):
I like the additional config flag options, but if you want to change the header besides adding your name, you need to do that in your fork.

Ragundah committed:
Fixed as requested

Ragundah commented:
Corrected and added name and update date under original header.

55Honey commented on LevelUpReward.lua:
```diff
@@ -37,17 +39,26 @@ Config_ItemAmount[35] = 10
-local Config_mailText = 2
-local senderGUID = 1
+-- General Settings Config
+local Config_mailText = 1           -- Which text to send in the mail to the player.
```
I want the default config to stay, so this needs to be 2

55Honey commented on LevelUpReward.lua:
```diff
@@ -37,17 +39,26 @@ Config_ItemAmount[35] = 10
+-- General Settings Config
+local Config_mailText = 1           -- Which text to send in the mail to the player.
+local Config_senderGUID = 1         -- GUID of the player found in CharactersSchema/Characters in Database.
+local Config_mailStationery = 61    -- Stationary used in the mail sent to the player. (41 Normal Mail, 61 GM/Blizzard Support, 62 Auction, 64 Valentines, 65 Christmas) Note: Use 62, 64, and 65 At your own risk.
```
same as above, must be 41 for the "Create a permanent copy of the letter" to work

Ragundah committed:
Requested changes for default config.

Ragundah commented:
Dont merge yet. An issue cropped up relating to crashing.
=

Ragundah committed:
Fixed crash due to SQL error.

Removed line Config_charactersDbName as no longer required.
Fixed crash due to SQL code trying to search for acore_characters.acore_characters.mail which does not exist.
Corrected checked = 2 from checked = 4 now correctly prevents returning (instead allows for deletion) to help with Mail DB clutter.
This also warns the player of deletion where returning has no warning.
Also modified the Mail1 code to include the senderGUID and provided information next to the config option advising how to change it to "From: Unknown" as the original did.

Ragundah commented:
Ready for any additional feedback/review as well as testing. Crash has been fixed. 62 64 and 65 stationary NOT linked to crashing it seems, but modifies the stationary in a way that could be undesirable. Leaving the text Note: Use 62, 64, and 65 At your own risk. for this reason.

Ragundah committed:
Mail Handling Changes

Changed from setting checked to prevent return to setting mail type to Creature to prevent return and reply.
Due to this players can use either mode to use player GUID or creature ID depending on the Config_preventReturn option.
Updated Config_senderGUID help note updated to reflect this.

55Honey commented on LevelUpReward.lua:
```diff
@@ -162,4 +181,4 @@ local function GrantReward(event, player, oldLevel)
 end
 
-RegisterPlayerEvent(PLAYER_EVENT_ON_LEVEL_CHANGE, GrantReward)
+RegisterPlayerEvent(PLAYER_EVENT_ON_LEVEL_CHANGE, GrantReward)
```
missing new line.

55Honey reviewed (changes requested):
Else looks good. I'll test it today.

Commit:
new line
